std/asyncio/net/Server.hx

```diff
@@ -0,0 +1,72 @@
+package asyncio.net;
+
+import asyncio.net.SocketOption.SocketOptionKind;
+import haxe.NoData;
+import haxe.Callback;
+import haxe.errors.NotImplemented;
+
+typedef ServerOptions = {
+	/**
+		Maximum size of incoming connections queue.
+		Default: 0
+		TODO: decide on a meaningful default value.
+	**/
+	var ?backlog:Int;
+	/**
+		Socket options as described in `asyncio.net.SocketOptions`
+	**/
+	var ?socketOptions:Array<SocketOption>;
+}
+
+/**
+	Server socket.
+**/
+class Server {
+	/**
+		Local address of this server.
+	**/
+	public var localAddress(get,never):SocketAddress;
+	function get_localAddress():SocketAddress throw new NotImplemented();
+
+	/**
+		Start a server on specified `address`.
+
+		This methods creates a socket, binds it to `address` and starts listening
+		for incoming connections.
+		Connections may be accepted with `server.accept` method.
+
+		Maximum size of incoming connections queue is specified by `options.backlog`.
+		If the queue is full, any new incoming connection will be rejected.
+	**/
+	static public function open(address:SocketAddress, ?options:ServerOptions, callback:Callback<Null<Server>>) {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Accept an incoming connection.
+	**/
+	public function accept(callback:Callback<Null<Socket>>) {
```

Comment:
So you have to poll even though it's asynchronous?

Reply:
What are the alternatives?
I was thinking about the ability to wrap the API with coroutines in future:
```haxe
while(true) {
  var client = server.accept.await();
  //...
}
```

Reply:
Well if it's a callback then it should just resolve with the connected socket when there is one, no?

Reply:
This `accept` does just that: once there's an incoming connection, `callback` is invoked.
I'm not sure what's your question.

Reply:
But the type of the callback is `Callback<Null<Socket>>`. Why would it ever be `null` then?

Reply:
In case of an error.

Reply:
Well, yes, but this is the case for every single callback at the moment, and none are marked with an explicit `Null` like this one.

Reply:
> none are marked with an explicit Null<…> like this one

Every callback is marked with explicit `Null` except `NoData` callbacks :)
See https://github.com/HaxeFoundation/haxe/pull/9111/files#r372352305 on why `Null` is not included in `CallbackHandler` signature for result.

Reply:
Oh, I just happened to see the primitive-taking callbacks (`Callback<Bool>`, `Callback<Int>`), which don't get wrapped. Disregard.